### PR TITLE
402 Enhance Zally Rules Command

### LINKED ISSUE #402: Enhance zally rules command
### Acceptance Criteria
1. If no parameters are set, `zally rules` returns the list of rules grouped by type
2. If additional parameter type is set, `zally rules` filters rules by type (API already has this feature)
3. If type is unknown, `zally rules` returns a warning

### PR DESCRIPTION
**Pull request by netme**
Closes #402

### CODE REVIEW

**Comment by netme** [comment]
@maschleg could you please check it once you'll have a moment?

File: cli-go/zally/utils/result_printer.go [hunk 1]
```diff
@@ -24,7 +24,14 @@ func NewResultPrinter(buffer io.Writer) ResultPrinter {
 
 // PrintRules prints a list of supported rules
 func (r *ResultPrinter) PrintRules(rules *domain.Rules) {
-	for _, rule := range rules.Rules {
+	r.printRules(rules.Must())
+	r.printRules(rules.Should())
+	r.printRules(rules.May())
+	r.printRules(rules.Hint())
```
Comment (maschleg) [inline cli-go/zally/utils/result_printer.go]: That looks inconvenient. Maybe have a sort function instead? Then we could sort by RuleType and RuleCode.

Comment (netme) [inline cli-go/zally/utils/result_printer.go]: @maschleg it also can be an option, however my approach allows to add headers for different rules sections later. Would you mind if I'll keep it as it is?

Comment (maschleg) [inline cli-go/zally/utils/result_printer.go]: ok

**Comment by maschleg** [comment]
looks good. Is it still under development?

**Comment by netme** [comment]
@maschleg , sorry, forgot to remove label. It suppose to be done :-)

**Comment by maschleg** [comment]
👍

**Pull request merged**